aws-logging-cloudwatch/src/main/java/com/amplifyframework/logging/cloudwatch/models/LoggingConstraints.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amplifyframework.logging.cloudwatch.models
+
+import com.amplifyframework.core.category.CategoryType
+import com.amplifyframework.logging.LogLevel
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * Constraints to control the log level per categoryType or user
+ */
+@Serializable
+data class LoggingConstraints @JvmOverloads constructor(
+    val defaultLogLevel: LogLevel = LogLevel.ERROR,
+    val categoryLogLevel: Map<CategoryType, LogLevel> = emptyMap(),
+    val userLogLevel: Map<String, UserLogLevel> = emptyMap(),
```

Comment:
It seems the userLogLevel overrides the root level categoryLogLevel, so why not design the types in such a way? 

Something like:
```
UserLoggingConstraints: LoggingConstraints {
    override val defaultLogLevel: LogLevel,
    override val categoryLogLevel: Map<CategoryType, LogLevel>
}
```